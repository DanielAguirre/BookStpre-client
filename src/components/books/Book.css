.book{
  flex-grow: 1;
  display: flex;
  margin-top: 20px;
}
  
.book-cover{
    width: 200px;
    height: 230px;
    background-image: url(img-dir'/no-cover-placeholder.jpg');
    background-size: cover;
}
.book-content {
  width: 75%;
  padding: 5%;
}
    