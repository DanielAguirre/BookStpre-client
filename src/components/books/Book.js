import React from 'react';
import glamorous from 'glamorous'
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

const Book =  glamorous.div({
  'flexGrow': 1,
  display: 'flex',
  'marginTop': '20px'
});

const Cover =  glamorous.div({
  width: '200px',
  height: '230px',
  backgroundSize: 'cover',
  backgroundImage: 'url(../img/no-cover-placeholder.jpg)'
});

const Content =  glamorous.div({
  width: '75%',
  padding: '5%'
});

const BookCard  = ({ book, getDate }) => (
  <Book className="book">
    <Cover className="book-cover">
      <img alt=''/>
    </Cover>
    <Content className="book-content" >
      <p className="book-title">{book.title}</p>
      <p className="book-author">{book.author}</p>
      <p className="book-price">{book.price}</p>
      <p className="book-publication">{book.publication}</p>
      <Link className="book-edit" to={`/edit/${book._id}`}>Edit </Link>
    </Content>
  </Book>
);

module.exports = BookCard;