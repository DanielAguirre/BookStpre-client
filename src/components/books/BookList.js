import React, { Component } from 'react';
import glamorous from 'glamorous'
import moment from 'moment';
import Book from './Book';

const Div =  glamorous.div({
  width: '100%',
  display: 'flex',
  boxSizing: 'border-box',
  flexWrap: 'wrap',
});

const BookList  = ({ bookList, getDate }) =>(
  <Div className="books-container">
    { bookList.map((book,i) => <Book book={book} key={book._id} /> )}
  </Div>
) 

export default BookList;