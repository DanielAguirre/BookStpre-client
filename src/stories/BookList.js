import React from 'react';
import { storiesOf, action, linkTo } from '@kadira/storybook';
import BookList from '../components/books/BookList';


storiesOf('BookList', module)
  .add('List of Books', () => (
    <BookList onClick={action('clicked')}>Hello Button</BookList>
  ))

