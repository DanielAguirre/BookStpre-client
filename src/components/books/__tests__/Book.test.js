import React from 'react';
import { shallow } from 'enzyme';
import Book from '../Book';
import BookList from '../BookList';

it('<Book />', () => {
  const book = {
    title: 'Harry Poterr 1',
    author: 'J.K. Rownling',
  };

  const component = shallow(<Book book={book}/>);
  expect(component).toMatchSnapshot();

})

it('<Books />', () => {
  const bookList = [{
    _id: 1,
    title: 'Harry Poterr 1',
    author: 'J.K. Rownling',
  }];

  const component = shallow(<BookList bookList={bookList}/>);
  expect(component).toMatchSnapshot();

})
