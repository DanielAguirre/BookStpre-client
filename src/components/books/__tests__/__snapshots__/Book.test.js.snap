exports[`test <Book /> 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <glamorous(div)
    className="book">
    <glamorous(div)
        className="book-cover">
        <img
            alt="" />
    </glamorous(div)>
    <glamorous(div)
        className="book-content">
        <p
            className="book-title">
            Harry Poterr 1
        </p>
        <p
            className="book-author">
            J.K. Rownling
        </p>
        <p
            className="book-price" />
        <p
            className="book-publication" />
        <Link
            className="book-edit"
            replace={false}
            to="/edit/undefined">
            Edit 
        </Link>
    </glamorous(div)>
</glamorous(div)>,
  "nodes": Array [
    <glamorous(div)
      className="book">
      <glamorous(div)
            className="book-cover">
            <img
                  alt="" />
      </glamorous(div)>
      <glamorous(div)
            className="book-content">
            <p
                  className="book-title">
                  Harry Poterr 1
            </p>
            <p
                  className="book-author">
                  J.K. Rownling
            </p>
            <p
                  className="book-price" />
            <p
                  className="book-publication" />
            <Link
                  className="book-edit"
                  replace={false}
                  to="/edit/undefined">
                  Edit 
            </Link>
      </glamorous(div)>
</glamorous(div)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BookCard
        book={
                Object {
                        "author": "J.K. Rownling",
                        "title": "Harry Poterr 1",
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "book": Object {
            "author": "J.K. Rownling",
            "title": "Harry Poterr 1",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <glamorous(div)
          className="book">
          <glamorous(div)
                    className="book-cover">
                    <img
                              alt="" />
          </glamorous(div)>
          <glamorous(div)
                    className="book-content">
                    <p
                              className="book-title">
                              Harry Poterr 1
                    </p>
                    <p
                              className="book-author">
                              J.K. Rownling
                    </p>
                    <p
                              className="book-price" />
                    <p
                              className="book-publication" />
                    <Link
                              className="book-edit"
                              replace={false}
                              to="/edit/undefined">
                              Edit 
                    </Link>
          </glamorous(div)>
</glamorous(div)>,
        "_debugID": 2,
        "_renderedOutput": <glamorous(div)
          className="book">
          <glamorous(div)
                    className="book-cover">
                    <img
                              alt="" />
          </glamorous(div)>
          <glamorous(div)
                    className="book-content">
                    <p
                              className="book-title">
                              Harry Poterr 1
                    </p>
                    <p
                              className="book-author">
                              J.K. Rownling
                    </p>
                    <p
                              className="book-price" />
                    <p
                              className="book-publication" />
                    <Link
                              className="book-edit"
                              replace={false}
                              to="/edit/undefined">
                              Edit 
                    </Link>
          </glamorous(div)>
</glamorous(div)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BookCard
    book={
        Object {
            "author": "J.K. Rownling",
            "title": "Harry Poterr 1",
          }
    } />,
}
`;

exports[`test <Books /> 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <glamorous(div)
    className="books-container">
    <BookCard
        book={
            Object {
                "_id": 1,
                "author": "J.K. Rownling",
                "title": "Harry Poterr 1",
              }
        } />
</glamorous(div)>,
  "nodes": Array [
    <glamorous(div)
      className="books-container">
      <BookCard
            book={
                  Object {
                        "_id": 1,
                        "author": "J.K. Rownling",
                        "title": "Harry Poterr 1",
                      }
            } />
</glamorous(div)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BookList
        bookList={
                Array [
                        Object {
                          "_id": 1,
                          "author": "J.K. Rownling",
                          "title": "Harry Poterr 1",
                        },
                      ]
        } />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bookList": Array [
            Object {
              "_id": 1,
              "author": "J.K. Rownling",
              "title": "Harry Poterr 1",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <glamorous(div)
          className="books-container">
          <BookCard
                    book={
                              Object {
                                        "_id": 1,
                                        "author": "J.K. Rownling",
                                        "title": "Harry Poterr 1",
                                      }
                    } />
</glamorous(div)>,
        "_debugID": 4,
        "_renderedOutput": <glamorous(div)
          className="books-container">
          <BookCard
                    book={
                              Object {
                                        "_id": 1,
                                        "author": "J.K. Rownling",
                                        "title": "Harry Poterr 1",
                                      }
                    } />
</glamorous(div)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BookList
    bookList={
        Array [
            Object {
              "_id": 1,
              "author": "J.K. Rownling",
              "title": "Harry Poterr 1",
            },
          ]
    } />,
}
`;
